src/routes/safe/components/Apps/components/SignMessageModal/ReviewMessage.tsx

```diff
@@ -0,0 +1,262 @@
+    isExecution,
+    gasCostFormatted,
+    txEstimationExecutionStatus,
+  } = useEstimateTransactionGas({
+    txData: txData || '',
+    txRecipient,
+    operation: Operation.DELEGATE,
+    txAmount: '0',
+    safeTxGas: manualSafeTxGas,
+    manualGasPrice,
+    manualGasLimit,
+  })
+
+  const [buttonStatus, setButtonStatus] = useEstimationStatus(txEstimationExecutionStatus)
+
+  const handleTxRejection = () => {
+    onTxReject(requestId)
+    onClose()
+  }
+
+  const handleUserConfirmation = (safeTxHash: string): void => {
+    onUserConfirm(safeTxHash, requestId)
+    onClose()
+  }
+
+  const confirmTransactions = (txParameters: TxParameters) => {
+    setButtonStatus(ButtonStatus.LOADING)
+
+    dispatch(
+      createTransaction(
+        {
+          safeAddress,
+          to: txRecipient,
+          valueInWei: '0',
+          txData,
+          operation: Operation.DELEGATE,
+          origin: app.id,
+          navigateToTransactionsTab: false,
+          txNonce: txParameters.safeNonce,
+          safeTxGas: txParameters.safeTxGas ? Number(txParameters.safeTxGas) : undefined,
+          ethParameters: txParameters,
+          notifiedTransaction: TX_NOTIFICATION_TYPES.STANDARD_TX,
+        },
+        handleUserConfirmation,
+        handleTxRejection,
+      ),
+    )
+
+    setButtonStatus(ButtonStatus.READY)
+  }
+
+  const closeEditModalCallback = (txParameters: TxParameters) => {
+    const oldGasPrice = Number(gasPriceFormatted)
+    const newGasPrice = Number(txParameters.ethGasPrice)
+    const oldSafeTxGas = Number(gasEstimation)
+    const newSafeTxGas = Number(txParameters.safeTxGas)
+
+    if (newGasPrice && oldGasPrice !== newGasPrice) {
+      setManualGasPrice(txParameters.ethGasPrice)
+    }
+
+    if (txParameters.ethGasLimit && gasLimit !== txParameters.ethGasLimit) {
+      setManualGasLimit(txParameters.ethGasLimit)
+    }
+
+    if (newSafeTxGas && oldSafeTxGas !== newSafeTxGas) {
+      setManualSafeTxGas(newSafeTxGas)
+    }
+  }
+
+  return (
+    <EditableTxParameters
+      ethGasLimit={gasLimit}
+      ethGasPrice={gasPriceFormatted}
+      safeTxGas={gasEstimation.toString()}
+      closeEditModalCallback={closeEditModalCallback}
+      isOffChainSignature={isOffChainSignature}
+      isExecution={isExecution}
+    >
+      {(txParameters, toggleEditMode) => (
+        <div>
+          <ModalTitle title={app.name} iconUrl={app.iconUrl} onClose={handleTxRejection} />
+
+          <Hairline />
+
+          <Container>
+            {/* Safe */}
+            <EthHashInfo name={safeName} hash={safeAddress} showAvatar showCopyBtn explorerUrl={explorerUrl} />
+            <StyledBlock>
+              <Text size="md">Balance:</Text>
+              <Text size="md" strong>{`${ethBalance} ${nativeCoin.symbol}`}</Text>
+            </StyledBlock>
+
+            <Divider withArrow />
+
+            <BasicTxInfo txRecipient={txRecipient} txData={txData} txValue="0" recipientName="SignMessageLib" />
```

Comment:
Is data in this case basically just the message itself? Or it encodes the addresses as well? Does it make sense to display it?

Reply:
No, it's encoded transaction data for SignMessageLib. We usually display transaction data for transactions, so I thought it makes sense to display it